### PR TITLE
Translator: es6ify 

### PR DESCRIPTION
**Pull request by yoavf**
This an attempt to es6ify the community translator jumpstart module. No changes in functionality expected. Happy to take suggestions for additional possible improvements on the syntax or code structure.

Test: Set ui locale to not EN, and enable the community translator option. Notice how the translator shows up and can be activated/deactivated.

https://calypso.live/?branch=fix/translator-es6

### CODE REVIEW

File: client/lib/translator-jumpstart/index.jsx [hunk 1]
```diff
@@ -1,37 +1,40 @@
...
+	communityTranslatorBaseUrl = 'https://widgets.wp.com/community-translator/',
+	communityTranslatorVersion = '1.160628';
 
 /**
  * Local variables
  */
-var communityTranslatorBaseUrl = 'https://widgets.wp.com/community-translator/',
-	communityTranslatorVersion = '1.160628',
-	translationDataFromPage = {
+var	translationDataFromPage = {
```
Comment (omarjackman) [inline client/lib/translator-jumpstart/index.jsx]: It looks like `translationDataFromPage` could also be a const

File: client/lib/translator-jumpstart/index.jsx [hunk 2]
```diff
@@ -69,12 +70,12 @@ communityTranslatorJumpstart = {
...
 	},
-	isActivated: function() {
+	isActivated() {
 		return _shouldWrapTranslations;
 	},
 
-	wrapTranslation: function( originalFromPage, displayedTranslationFromPage, optionsFromPage ) {
-		var props, dataElement;
+	wrapTranslation( originalFromPage, displayedTranslationFromPage, optionsFromPage ) {
+		var props;
```
Comment (omarjackman) [inline client/lib/translator-jumpstart/index.jsx]: This could also be a const
```js
const props = { className: 'translatable' };
```

Comment (omarjackman) [inline client/lib/translator-jumpstart/index.jsx]: also noting that the logic from lines 87 to 94 could be simplified if we return early

```js
if ( 'string' !== typeof originalFromPage ) {
	debug( 'unknown original format' );
 	return displayedTranslationFromPage;
}

const props = {
	className: 'translatable',
	'data-singular': originalFromPage
};
```

File: client/lib/translator-jumpstart/index.jsx [hunk 3]
```diff
@@ -104,21 +105,21 @@ communityTranslatorJumpstart = {
 		}
 
 		// React.DOM.data returns a frozen object, therefore we make a copy so that we can modify it below
-		dataElement = Object.assign( {}, React.DOM.data( props, displayedTranslationFromPage ) );
+		const dataElement = Object.assign( {}, React.DOM.data( props, displayedTranslationFromPage ) );
 
 		// now we can override the toString function which would otherwise return [object Object]
 		dataElement.toString = function() {
```
Comment (omarjackman) [inline client/lib/translator-jumpstart/index.jsx]: could be:
```js
dataElement.toString = () => displayedTranslationFromPage;
```

**Comment by yoavf** [comment]
Thanks @omarjackman for the review!

File: client/lib/translator-jumpstart/index.js [hunk 4]
```diff
@@ -161,15 +166,15 @@ communityTranslatorJumpstart = {
...
+			translationDataFromPage.pluralForms;
 		translationDataFromPage.currentUserId = user.data.ID;
 
-		// extract the active language's name out of Calypso lanaguage data
-		for ( i = 0; i < languages.length; i++ ) {
-			if ( languages[ i ].langSlug === localeCode ) {
-				translationDataFromPage.languageName = languages[ i ].name.replace( /^(?:[a-z]{2,3}|[a-z]{2}-[a-z]{2})\s+-\s+/, '' );
-				break;
-			}
+		const currentLocale = languages.find( lang => lang.langSlug === localeCode );
```
Comment (akirk) [inline client/lib/translator-jumpstart/index.js]: Nicely shortened section :)

File: client/lib/translator-jumpstart/index.js [hunk 5]
```diff
@@ -84,7 +84,15 @@ communityTranslatorJumpstart = {
...
 
-		props = { className: 'translatable' };
+		if ( 'string' !== typeof originalFromPage ) {
+			debug( 'unknown original format' );
+			return displayedTranslationFromPage;
+		}
+
+		const props = {
+			className: 'translatable',
+			'data-singular': originalFromPage
```
Comment (akirk) [inline client/lib/translator-jumpstart/index.js]: Intended change of functionality?

File: client/lib/translator-jumpstart/index.js [hunk 6]
```diff
@@ -84,7 +84,15 @@ communityTranslatorJumpstart = {
...
+		};
 
 		if ( 'string' === typeof originalFromPage ) {
```
Comment (akirk) [inline client/lib/translator-jumpstart/index.js]: If yes, then this can be removed.

File: client/lib/translator-jumpstart/index.js (hunk 5 above)
Comment (akirk) [inline client/lib/translator-jumpstart/index.js]: I am not sure we can do this. This is only when the page provides the original and we don't query the translate.wordpress.com API for the string, right?

Comment (yoavf) [inline client/lib/translator-jumpstart/index.js]: There was no change in functionality, but the diff shown here was missing something from the current version. See here: https://github.com/Automattic/wp-calypso/blob/master/client/lib/translator-jumpstart/index.js#L89

But there was some duplicate code that I've now removed.

**Comment by akirk** [comment]
Tested that everything works. LGTM!

**Pull request merged**